public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,903 @@
+  and values of the HTML controls on screen.
+
+  With _reactive_ forms, you create a tree of Angular form control objects
+  in the component class and bind them to native form control elements in the 
+  component template, using techniques described in this guide.
+
+  With reactive forms, you create and manipulate form control objects directly in the 
+  component class. As the component class has immediate access to both the data 
+  model and the form control structure, you can push data model values into 
+  the form controls and pull user-changed values back out. The component can 
+  observe changes in form control state and react to those changes.
+
+  One advantage of working with form control objects directly is that value and validity updates 
+  are always synchronous and under your control. 
+  You won't encounter the timing issues that sometimes plague a template-driven form.
+
+  In keeping with the reactive paradigm, the component 
+  preserves the immutability of the _data model_,
+  treating it as a pure source of original values.
+  Rather than update the data model directly, 		
+  the component extracts user changes and forwards them to an external component or service,		
+  which does something with them (such as saving them) 		
+  and returns a new _data model_ to the component that reflects the updated model state.		
+    
+  Using reactive form directives does not require you to follow all reactive priniciples,		
+  but it does facilitate the reactive programming approach should you choose to use it.
+
+  ### _Template-driven_ forms
+
+  _Template-driven_ forms, introduced in the [Template guide](forms.html), take a completely different approach.
+  
+  You place HTML form controls (such as `<input>` and `<select>`) in the component template and 
+  bind them to _data model_ properties in the component, using directives 
+  like `[(ngModel)]`.
+
+  You don't create Angular form control objects. Angular directives 
+  create them for you, using the information in your data bindings. 
+  You don't push and pull data values. Angular handles that for you with `ngModel`.
+  Angular updates the mutable _data model_ with user changes as they happen.
+
+  When writing in a reactive style, you determine the timing and flow of data between 
+  the data model and the screen. You rarely, if ever, use two-way data binding. 
+  For this reason, the ngModel directive is not part of the ReactiveFormsModule.
+  
+  These differences reflect two architectural paradigms, 
+  with their own strengths and weaknesses,
+  and you are free to choose between them.
+
+  The balance of this _reactive forms_ guide assumes the _reactive_ paradigm and 
+  concentrates exclusively on the reactive forms techniques. For 
+  information on _template driven forms_, see the [Template Guide](forms.html).
+
+  You'll learn about reactive forms by building one from scratch.
+
+  In the next section, you'll prepare your project for reactive forms.
```

Comment:
Can we say 'set up your project for the reactive form demo' or something similar?  What's here implies that these steps are all necessary, when some steps like creating the Hero model, etc, are demo-specific.